src/main/scala/mesosphere/marathon/core/storage/store/impl/cache/LazyCachingPersistenceStore.scala

```diff
@@ -101,8 +107,13 @@ class LazyCachingPersistenceStore[K, Category, Serialized](
           Future.successful(v)
         case _ =>
           async { // linter:ignore UnnecessaryElseBranch
-            val value = await(store.get(id))
+            val value: Option[V] = await(store.get(id))
             valueCache.put(storageId, value)
+            value.foreach { v =>
+              if (ir.hasVersions) {
+                versionedValueCache.put((storageId, ir.version(v)), value)
```

Comment:
Shouldn't this also put the version into the list of versions in the versionCache?